daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -97,8 +97,9 @@ private[lf] object Speedy {
 
   private type Actuals = util.ArrayList[SValue]
 
-  /** The speedy CEK machine. */
-  final class Machine(
+  private[lf] sealed trait LedgerMode
```

Comment:
Very open to better naming suggestions.

Reply:
I do not like it neither, but I do not have better alternative. :disappointed:

Reply:
What do you think about something like `ExecutionEnvironment`? You can either execute DAML on a ledger or off a ledger.

Reply:
sounds reasonable but also sounds very generic to the point where it doesn’t give you much of an idea of what the possible execution environments could be.